test/integration/ext/openai/model.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+import vcr
+
+from superduperdb.ext.openai.model import OpenAIChatCompletion, OpenAIEmbedding
+
+
+@vcr.use_cassette(
+    'test/integration/ext/openai/cassettes/test_embed.yaml',
+    filter_headers=['authorization'],
+)
+def test_embed():
+    e = OpenAIEmbedding(model='text-embedding-ada-002')
+    resp = e.predict('Hello, world!')
+
+    assert len(resp) == e.shape[0]
+
+    assert all([isinstance(x, float) for x in resp])
```

Comment:
Similar:

`assert all(-1 <= x <= 1 for x in resp))`

Also, you don't just not need the square brackets, it makes it on average slower and use more memory!